Dockerfile

```diff
@@ -6,17 +6,13 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get -q update && \
     apt-get -q --yes install \
-      mysql-client \
-      nodejs-legacy \
-      npm && \
+      mysql-client && \
     apt-get clean
 
 WORKDIR /app
 
-COPY . /app
-RUN python setup.py install
-
-WORKDIR /app/ui
-RUN npm install
-
-WORKDIR /app
+# Copying Balrog to /app instead of installing it means that production can run
+# it, and we can bind mount to override it for local development.
+COPY auslib requirements.txt setup.py ui uwsgi version.json /app/
+# Add --require-hashes after python2.6 support is dropped, and uncomment everything in requirements.txt
+RUN pip install -r requirements.txt
```

Comment:
I'm back/forth on this Dockerflow requirement as well. It's there so the container can be treated like any linux executable. 

how about: 

run.sh
```
#!/bin/sh

if [ $1 == "public" ]; then
   exec <cmd to run uswgi public>
elif [ $1 == "admin" ]; then
   exec <cmd to run uwsgi admin>
else
   echo "unknown mode: $1"
   exit 1
fi
```

The `exec` is important as it will replace the shell with another program. This means signals `SIGHUP`, `SIGKILL`, etc will pass to uwsgi correctly.

```

In the Dockerfile, the default is to run the public facing app
...
ENTRYPOINT ["/app/bin/run.sh"]
CMD ["public"]
```

So to run the container:
public: `docker run <options> balrog:latest`
admin: `docker run <options> balrog:latest admin`